deployments/helm-chart/values.yaml

```diff
@@ -474,8 +461,26 @@ prometheus:
     ## Requires prometheus.create=true
     create: false
 
+    ## Name of the endpoint exposed by the service.
+    ## Used by the ServiceMonitor's `endpoints` field to discover service endpoints.
+    endpoint: "prometheus"
+
     labels: {}
 
+  serviceMonitor:
+    ## Creates a serviceMonitor to expose statistics on the kubernetes pods.
+    create: false
+
+    ## Kubernetes object labels to attach to the serviceMonitor object.
+    labels: {}
+
+    ## A set of labels to allow the selection of endpoints for the ServiceMonitor.
+    selectorMatchLabels: {}
+
+    ## A list of endpoints allowed as part of this ServiceMonitor.
+    endpoints: []
+    # - port: prometheus
+
```

Comment:
I was doing some testing with this. As is, with the `- port: prometheus` commented out, metric scrapping will not work. (prometheus dashboard will not pick up the servicemonitor)

The service will get generated as the following:

```
spec:
  endpoints: []
  selector:
    matchLabels:
      app: nic-servicemonitor
      app.kubernetes.io/instance: test01
      app.kubernetes.io/name: nginx-ingress
```

Would it make more sense to just hard code these values in and we can generate them upon install for the users?

Reply:
> I was doing some testing with this. As is, with the `- port: prometheus` commented out, metric scrapping will not work. (prometheus dashboard will not pick up the servicemonitor)

This tells me that:
1 - this needs to be uncommented with a note to not change it ( values.prometheus.serviceMonitor.endpoints )
2 - also needs a note not to change it "prometheus" line ( values,prometheus.endpoint )

This will let us get this out the door with minimal change

In a future release we need to hide these two options.  I really don't see any value in these fields being configurable as it really just sets someone up to break their monitor.

Reply:
since we're moving all the fields in this PR (and this hasn't been released yet), I think we can remove the `endpoints` as part of the changes here.